src/main/java/ca/uhn/fhir/jpa/starter/CollectDataResourceProvider.java

```diff
@@ -0,0 +1,155 @@
+package ca.uhn.fhir.jpa.starter;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import com.google.common.base.Charsets;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.IdType;
+import org.hl7.fhir.r4.model.Measure;
+import org.hl7.fhir.r4.model.MeasureReport;
+import org.hl7.fhir.r4.model.Observation;
+import org.hl7.fhir.r4.model.Parameters;
+import org.hl7.fhir.r4.model.Reference;
+import org.hl7.fhir.r4.model.Resource;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.jpa.rp.r4.MeasureResourceProvider;
+import ca.uhn.fhir.rest.annotation.IdParam;
+import ca.uhn.fhir.rest.annotation.Operation;
+import ca.uhn.fhir.rest.annotation.OperationParam;
+import ca.uhn.fhir.rest.api.Constants;
+import ca.uhn.fhir.rest.api.EncodingEnum;
+
+public class CollectDataResourceProvider {
+    protected static CloseableHttpClient ourHttpClient;
+
+    @Autowired
+    MeasureResourceProvider measureResourceProvider;
+
+    FhirContext ctx = FhirContext.forR4();
+
+    @Operation(name = PlirConstants.OPERATION_COLLECT_DATA, idempotent = true, type = Measure.class)
+    public Parameters collectDataOperation(HttpServletRequest theServletRequest, @IdParam IdType theId,
```

Comment:
See [the spec](http://hl7.org/fhir/R4/measure-operation-collect-data.html). This parameter is supposed to take a reference rather than *just* the id.

```suggestion
    public Parameters collectDataOperation(HttpServletRequest theServletRequest, @OperationParam(name = "measure", min = 1, max = 1) Reference measureReference,
```

Reply:
@ibacher am not getting your comment here.

see format of our get request
`GET [base]/Measure/<ID>/$collect-data?periodStart=2018&periodEnd=201` .
The only request paramers we are passing are `periodStart` and `periodEnd`
we are not passing a parameter `measure`  ,unless you suggesting we change the request to something like

`GET [base]/Measure/TX-PVLS/$collect-data?periodStart=2018&periodEnd=2011&measure=someReference` 

Am not using `@IdParam IdType theId`  as a request Parameter , but it simply refers to the resourceId of the Resurce in the request in our case to  `/TX-PVLS/`.
When i looked throught the HAPI API , thats how the resource id captured.

Reply:
Oh, I see, makes sense.

Reply:
Alright